gems/aws-sdk-route53/lib/aws-sdk-route53/plugins/id_fix.rb

```diff
@@ -29,7 +29,7 @@ def remove_id_prefixes(params)
             # Many operations accept of :id or :hosted_zone_id as a root-level
             # param, pruning prefixes from those.
             [:id, :hosted_zone_id, :delegation_set_id].each do |key|
-              params[key] = remove_prefix(params[key]) if params[key]
+              params[key] = remove_prefix(params[key]) if params.to_hash.key?(key)
```

Comment:
how about using `to_h` of [Aws::Structure](https://github.com/aws/aws-sdk-ruby/blob/master/gems/aws-sdk-core/lib/aws-sdk-core/structure.rb#L29) : )

Reply:
I had to rollback the change since it broke the build using Ruby 1.9.3 https://travis-ci.org/aws/aws-sdk-ruby/builds/398037219.

After doing some investigation in the Aws::Structure , I found out that the Aws::Structure#to_h has an alias https://github.com/aws/aws-sdk-ruby/blob/master/gems/aws-sdk-core/lib/aws-sdk-core/structure.rb#L46 . So both #to_hash and #to_h point to the same method in the Aws::Structure class.